### PR TITLE
Readme for Nix

### PR DESCRIPTION
**Pull request by querolita**
This is a readme file specifically for the Nix building method. It is not meant to be a final version whatsoever, but to provide some initial guidelines for those who are facing issues when using Nix. It contains some basic instructions, followed by common errors that I have encountered myself, with proposals of how to fix them (at least on my M1). If it is not sufficiently generic to be merged, I will happily leave it as a draft PR or an internal Notion page for the onboarding process.

Special thanks to @dannywillems , @Trivo25 , @hattyhattington17 , @georgeee , @Geometer1729, and @mrmr1993, for your inputs and support.

### CODE REVIEW

File: README-nix.md [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+```console
+error: "/nix/store/w30zw23kmgks77d870i502a3185hjycv-rust/lib/rustlib/src/rust/Cargo.lock"
+       does not exist, unable to build with the standard library, try:
+                   rustup component add rust-src --toolchain nix
+```
+
+#### Fix
+
+Install `cargo` on your host machine. If doing so does not solve the problem, it
+is very likely that the whole Nix setup has gone wrong and it is very adviceable
```
Comment (45930) [inline README-nix.md:253]:
```suggestion
is very likely that the whole Nix setup has gone wrong and it is very advisable
```

File: README-nix.md [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+to install it from scratch.
+
+### Old path
+
+When several clones of the repository are present in the system and both have
+used Nix (or if it has been moved from one location to another), Nix might cache
+an old path, impossibilitating builds. For example, typing `nix develop mina`
```
Comment (45930) [inline README-nix.md:260]:
```suggestion
an old path, breaking builds. For example, typing `nix develop mina`
```

File: README-nix.md [hunk 3]
```diff
@@ -0,0 +1,271 @@
...
+
+```console
+error: resolving Git reference 'master': revspec 'master' not found
+```
+
+Then, the error message would still contain old directories.
+
+#### Fix
+
+Create a new environment for Nix and start from scratch.
```
Comment (45930) [inline README-nix.md:271]: How?

**Review by Geometer1729 — APPROVED** [review]
This is great, sorry I wasn't much help with the MacOS stuff

File: README-nix.md [hunk 4]
```diff
@@ -0,0 +1,271 @@
+# o1js README-nix
+
+[Nix](https://nixos.org/) is a tool for package management and system
+configurationthat can help developers build a project in a reproducible and
```
Comment (dannywillems) [inline README-nix.md:4]:
```suggestion
configuration that can help developers build a project in a reproducible and
```

File: README-nix.md [hunk 5]
```diff
@@ -0,0 +1,271 @@
...
+uninstalling and reinstalling Nix from scratch.
+
+> **warning for macOS users**: macOS updates will often break your nix
+> installation. To prevent that, you can add the following to your `~/.bashrc`
+> or `~/.zshrc`:
+>
+> ```bash
+> # avoid macOS updates to destroy nix
+> if [ -e '/nix/var/nix/profiles/default/etc/profile.d/>nix-daemon.sh' ]; then
+>  source '/nix/var/nix/profiles/default/etc/profile.d/nix-daemon.sh'
```
[the remark this answers is not in the document]
Comment (hattyhattington17) [reply README-nix.md:35]: I am not sure why this is the fix, but I've seen online that OS updates will cause nix to break on macOS. I would like to try this installer in the future on a clean machine and see if it lets us avoid this step https://determinate.systems/posts/nix-survival-mode-on-macos/

**Review by Trivo25 — APPROVED** [review]
I am good with merging this. We could add a disclaimer that not all of these "solutions" might work universally and might vary depending on your environment, but I would rather provide hints than not talk about it at all - thanks! :)

**Pull request merged**